docs/docs/ScalaUserGuide/build-bigdl-core.md

```diff
@@ -0,0 +1,125 @@
+## **Build BigDL-core on Different Platforms**
+
+### **Setup Build Environment**
+
+For building BigDL-core, there should have
+
++ JDK 1.7+
++ maven
++ make
++ g++-7
++ Intel Parallel Studio
++ Git.
+
+BigDL-core is a JNI project, `mkl2017-xeon-blas` needs MKL libraries with icc and `bigquant` needs g++-7. We use `maven` + `make` to control the build process where maven for java and make for c/c++ code.
+
+### **Environment on CentOS**
+
+* **Build GCC-7.2**
+
+```bash
+git clone https://github.com/gcc-mirror/gcc.git
+git checkout gcc-7_2_0-release
+./configure --prefix=/opt/gcc-7.2.0 --enable-languages=c,c++ --disable-multilib --disable-nls
+make -j4 && make install
+ln -s /opt/gcc-7.2.0 /opt/gcc
+```
+
+* **Build binutils 2.29**
+
+```bash
+wget https://ftp.gnu.org/gnu/binutils/binutils-2.29.tar.gz
+tar zxvf binutils-2.29.tar.gz -C /tmp/ && cd /tmp/binutils-2.29
+configure --prefix=/opt/binutils-2.29
+make && make install
+ln -s /opt/binutils-2.29 /opt/binutils/
+```
+
+* **Install Git**
+
+```bash
+./configure --prefix=/opt/git-2.9.5
+make -j4 && make install
+ln -s /opt/git-2.9.5 /opt/git
+```
+
+* **Install Intel Parallel Studio XE**
```

Comment:
any instruction to install intel parallel sutdio?

Reply:
Installing Intel Parallel Studio XE is very simple. There's a GUI. You need only to click the *next* to the end. It will automatically install.

Reply:
Add an link to parallel studio?

Reply:
OK.